### PR TITLE
Masking tasks for GNN Modular

### PR DESCRIPTION
**Pull request by tonydavis629**
## Description

Fix #(issue)

Implements atom and edge masking tasks for GNNModular. Also implements an EgoGraphFeaturizer which is used for masking tasks in Strategies for Graph Pretraining by Hu et al. 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/losses.py [hunk 1]
```diff
@@ -976,6 +976,40 @@ def loss(node_emb, inputs):
         return loss
 
 
+class GraphNodeMaskingLoss(Loss):
```
Comment (rbharath) [inline deepchem/models/losses.py:979]: Can you add docstring with usage example?

File: deepchem/models/losses.py [hunk 2]
```diff
@@ -976,6 +976,40 @@ def loss(node_emb, inputs):
...
+
+            if self.mask_edge:
+                loss += self.criterion(pred_edge,
+                                       inputs.mask_edge_label[:, 0].long())
+            return loss
+
+        return loss
+
+
+class GraphEdgeMaskingLoss(Loss):
```
Comment (rbharath) [inline deepchem/models/losses.py:1000]: Same here

File: deepchem/models/losses.py [hunk 3]
```diff
@@ -976,6 +976,40 @@ def loss(node_emb, inputs):
         return loss
```
Comment (rbharath) [inline deepchem/models/losses.py:978]: Can you add the new losses to `losses.rst`?

File: deepchem/models/torch_models/gnn.py [hunk 4]
```diff
@@ -296,58 +316,73 @@ def build_components(self):
         encoders = torch.nn.ModuleList(encoders)
         batch_norms = torch.nn.ModuleList(batch_norms)
 
-        if self.graph_pooling == "sum":
```
Comment (rbharath) [inline deepchem/models/torch_models/gnn.py:299]: Has this code been moved elsewhere?

File: deepchem/models/torch_models/gnn.py [hunk 5]
```diff
@@ -390,6 +429,34 @@ def classification_loss(self, inputs, labels):
         class_loss = self.criterion(out, labels)
         return class_loss
 
+    def masked_node_loss(self, inputs, labels):
```
Comment (rbharath) [inline deepchem/models/torch_models/gnn.py:432]: Can you add a docstring?

**Review by rbharath — COMMENTED** [review]
Requests for more comments/docstrings below. I will do a second review pass once those are in

File: deepchem/models/torch_models/gnn.py (hunk 4 above)
Comment (tonydavis629) [reply deepchem/models/torch_models/gnn.py:299]: Yes, line 348

**Review by rbharath — COMMENTED** [review]
Nothing major jumping out I think. Can you respond to the open comments and I'll do a final review?

File: deepchem/models/losses.py (hunk 3 above)
Comment (tonydavis629) [reply deepchem/models/losses.py:978]: done

File: deepchem/models/losses.py (hunk 1 above)
Comment (tonydavis629) [reply deepchem/models/losses.py:979]: done

File: deepchem/models/losses.py (hunk 2 above)
Comment (tonydavis629) [reply deepchem/models/losses.py:1051]: done

**Review by rbharath — APPROVED** [review]
LGTM
